sw/device/silicon_creator/lib/drivers/pinmux.c

```diff
@@ -0,0 +1,69 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/drivers/pinmux.h"
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/silicon_creator/lib/base/abs_mmio.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "pinmux_regs.h"  // Generated.
+
+/**
+ * A peripheral input and MIO pad to link it to.
+ */
+typedef struct pinmux_input {
+  top_earlgrey_pinmux_peripheral_in_t periph;
+  top_earlgrey_pinmux_insel_t pad;
+} pinmux_input_t;
+
+/**
+ * Peripheral input to MIO pad mappings.
+ */
+static const pinmux_input_t kPinmuxInputs[] = {
+    /**
+     * UART
+     */
+    {.periph = kTopEarlgreyPinmuxPeripheralInUart0Rx,
+     .pad = kTopEarlgreyPinmuxInselIoc10},
+};
```

Comment:
FYI: The mask_rom UART driver does not have code to read from the receive FIFO.

I don't object to having a pin defined as the UART RX input - not having this defined resulted in confusion and divergent designs with a different chip.

Reply:
Ah I didn't realise that but it seems obvious in retrospect. I think we should probably remove it but I don't mind keeping it if there is a benefit. I'll probably wait until we add another input though as otherwise the input configuration code will be dead.